src/geometry/euclidean/oriented_angle.lean

```diff
@@ -773,3 +780,424 @@ begin
 end
 
 end orthonormal
+
+namespace orientation
+
+open finite_dimensional
+
+variables {V : Type*} [inner_product_space ℝ V]
+variables [hd2 : fact (finrank ℝ V = 2)] (o : orientation ℝ V (fin 2))
+include hd2 o
```

Comment:
Did you consider just using a plain `structure`?

E.g., something like:
```lean
variables (V : Type*) [inner_product_space ℝ V]

structure oriented_space (d : ℕ) :=
(orientation : orientation ℝ V (fin d))
(dim_eq : finrank ℝ V = d)

variables (o : oriented_space V 2)

local notation `ob` := o.orientation.fin_orthonormal_basis_orthonormal dec_trivial o.dim_eq
```

Reply:
I didn't consider it, but I think it gets more awkward once you start talking about multiple orientations of the same space, or constructing an orientation in some way other than having a variable for it sitting around, since all the operations for manipulating and constructing orientations work with `orientation` not `oriented_space`.

I expect most uses of oriented angles in the end to work with the third version of the API (not yet implemented), the one for Euclidean affine spaces, where the orientation is implicit through a `module.oriented` instance (thus, defined by applying this version of the API to the `positive_orientation`) and the angles use the notation `∡ A B C` to refer to the oriented angle given by three points (cf. `∠ A B C` for the unoriented angle).

Reply:
I'm happy to take your word for this. In any case the current design is certainly solid.